src/main/Extensions/CustomWebSearch/CustomWebSearchExtension.ts

```diff
@@ -0,0 +1,135 @@
+import type { AssetPathResolver } from "@Core/AssetPathResolver";
+import type { Extension } from "@Core/Extension";
+import type { SettingsManager } from "@Core/SettingsManager";
+import type { OperatingSystem, SearchResultItem } from "@common/Core";
+import type { Image } from "@common/Core/Image";
+import type { CustomSearchEngineSetting, Settings } from "@common/Extensions/CustomWebSearch";
+
+export class CustomWebSearchExtension implements Extension {
+    public readonly id = "CustomWebSearch";
+    public readonly name = "Browser Search";
+
+    public readonly nameTranslation = {
+        key: "extensionName",
+        namespace: "extension[CustomWebSearch]",
+    };
+
+    public readonly author = {
+        name: "NiewView",
+        githubUserName: "NiewView",
+    };
+
+    public constructor(
+        private readonly operatingSystem: OperatingSystem,
+        private readonly assetPathResolver: AssetPathResolver,
+        private readonly settingsManager: SettingsManager,
+    ) {}
+
+    async getSearchResultItems(): Promise<SearchResultItem[]> {
+        // Custom search engines do not have static search results
+        return [];
+    }
+
+    public getInstantSearchResultItems(searchTerm: string): SearchResultItem[] {
+        const customSearchEngines = this.settingsManager.getValue<CustomSearchEngineSetting[]>(
+            `extension[${this.id}].customSearchEngines`,
+            this.getSettingDefaultValue("customSearchEngines"),
+        );
```

Comment:
You can use `getExtensionSettingKey` from `@common/Core/Extension` to get the right setting key:
 
```suggestion
        const customSearchEngines = this.settingsManager.getValue<CustomSearchEngineSetting[]>(
            getExtensionSettingKey(this.id, "customSearchEngines"),
            this.getSettingDefaultValue("customSearchEngines"),
        );
```